Intersect.Client/Interface/Game/Map/MinimapWindow.cs

```diff
@@ -0,0 +1,571 @@
+
+            if (_entityCache.TryGetValue(position, out var cachedEntity))
+            {
+                Graphics.Renderer.DrawTexture(cachedEntity, 0, 0, cachedEntity.Width,
+                    cachedEntity.Height, x, y, cachedEntity.Width, cachedEntity.Height,
+                    Color.White, _renderTexture);
+            }
+        }
+
+        private static Dictionary<MapPosition, MapBase> CreateMapGridFromMap(MapInstance map)
+        {
+            var grid = new Dictionary<MapPosition, MapBase>();
+            for (var x = map.GridX - 1; x <= map.GridX + 1; x++)
+            {
+                for (var y = map.GridY - 1; y <= map.GridY + 1; y++)
+                {
+                    if (x < 0 || x >= Globals.MapGridWidth ||
+                        y < 0 || y >= Globals.MapGridHeight ||
+                        Globals.MapGrid[x, y] == Guid.Empty)
+                    {
+                        continue;
+                    }
+
+                    int minimapX = x - (map.GridX - 1);
+                    int minimapY = y - (map.GridY - 1);
+
+                    grid.Add((MapPosition)(minimapX + (minimapY * 3)), MapBase.Get(Globals.MapGrid[x, y]));
+                }
+            }
+            
+            return grid;
+        }
+
+        private Dictionary<MapPosition, Dictionary<Point, EntityInfo>> GenerateEntityInfo(Dictionary<Guid, Entity> entities, Player player)
+        {
+            var dict = new Dictionary<MapPosition, Dictionary<Point, EntityInfo>>();
+
+            foreach (var entity in entities.Values)
+            {
+                if (!_mapPosition.TryGetValue(entity.MapInstance.Id, out var map))
+                {
+                    continue;
+                }
+
+                if (entity.IsHidden)
+                {
+                    continue;
+                }
+
+                var color = Color.Transparent;
+                var texture = string.Empty;
+                
+                if (entity.Id == player.Id)
+                {
+                    color = Options.Instance.MinimapOpts.MinimapColors.MyEntity;
+                    texture = Options.Instance.MinimapOpts.MinimapImages.MyEntity;
+                }
+                else
+                {
+                    switch (entity.Type)
+                    {
+                        case EntityType.Player:
+                            if (player.IsInMyParty(entity.Id))
+                            {
+                                color = Options.Instance.MinimapOpts.MinimapColors.PartyMember;
+                                texture = Options.Instance.MinimapOpts.MinimapImages.PartyMember;
+                            }
+                            else
+                            {
+                                color = Options.Instance.MinimapOpts.MinimapColors.Player;
+                                texture = Options.Instance.MinimapOpts.MinimapImages.Player;
+                            }
+
+                            break;
```

Comment:
Move the `Id` check into the `EntityType.Player` switch case. `Id` is _not safe_ to compare between different types.